clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/StateRepositoryOpCodec.java

```diff
@@ -152,8 +155,26 @@ public StateRepositoryOpMessage decode(EhcacheMessageType messageType, ByteBuffe
     String mapId = decoder.string(MAP_ID_FIELD);
 
     ByteBuffer keyBuffer = decoder.byteBuffer(KEY_FIELD);
-    Object key = Util.unmarshall(keyBuffer);
+    Object key = Util.unmarshall(keyBuffer, WHITELIST_PREDICATE);
 
     return new StateRepositoryOpMessage.GetMessage(storeName, mapId, key);
   }
+
+  public static final Predicate<Class<?>> WHITELIST_PREDICATE = Collections.unmodifiableList(Arrays.asList(
+    java.lang.Integer.class,
```

Comment:
List contains is O(N)... we should really be using a set contains here.  Also the unmodifiable is unnecessary if you're only retaining the object through the predicate

Reply:
changed to hashset